src/core/eval.ml

```diff
@@ -407,6 +407,16 @@ let rec hnf : ctxt -> term -> term = fun ctx t ->
      Abst(a, Bindlib.unbox (Bindlib.bind_var x (lift (hnf ctx t))))
   | t         -> t
 
+(** [type_app a ts] returns the type of [add_args x ts] where [x] is any
```

Comment:
I suggest to move this function to infer.ml if that is possible.

Reply:
fixed